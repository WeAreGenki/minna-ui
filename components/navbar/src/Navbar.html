<!--
  NAVBAR COMPONENT

  A simple fixed top navigation bar with a floating effect upon scrolling. It
  will collapse into a hamburger menu when the screen width is small.

  USAGE:
    The "menuItems" property should be an array of objects which look like this:

      <Navbar current="{child.segment}" menuItems="{[
        { url: '/page-one', text: 'Page One' },
        { url: '/page-two', text: 'Page Two' },
        { url: '/about', text: 'About Us' },
      ]}"/>

    The menu hamburger, menu close, and logo images are referenced using the
    inline SVG sprite with <symbol> + <use> elements technique. Because these
    images are critical to page rendering, we use this technique as it's the
    faster way to load UI critical SVGs. This component doesn't come with any
    SVG images so you need to provide your own. For example, you could include
    something structured like this after the opening <body> tag in your page:

      <svg class="dn" xmlns="http://www.w3.org/2000/svg">
        <defs>
          <symbol id="logo" viewBox="0 0 67.2 25.6">
            YOUR LOGO SVG CONTENTS
          </symbol>
          <symbol id="menu" class="icon" viewBox="0 0 24 24">
            <path d="M3 12h18M3 6h18M3 18h18" />
          </symbol>
          <symbol id="x" class="icon" viewBox="0 0 24 24">
            <path d="M18 6L6 18M6 6l12 12" />
          </symbol>
        </defs>
      </svg>

    Also add this to your CSS:

      .icon {
        fill: none;
        stroke: currentColor;
        stroke-linecap: round;
        stroke-linejoin: round;
        stroke-width: 2;
      }

    You may need to adjust size and whitespace to suite your logo. See the CSS
    variables below.

    Alternatively, you may replace the SVG logo with your own markup by passing
    the markup via a slot, that is, by adding it in between Navbar tags:

      <Navbar menuItems="{[...]}">
        <span class="navbar-logo">YOUR LOGO</span>
      </Navbar>

  NOTE:
    In the code, object properties starting with _ are safe to mangle; the name
    can be shortened at build time for smaller JavaScript bundles.

-->

<header class="navbar{_hasScrolled || _isOpen ? ' navbar-active' : ''}">
  <nav class="dfc fww con" role="navigation">
    <button
      type="button"
      class="navbar-button l-dn button-clear"
      aria-label="menu toggle"
      aria-expanded="{_isOpen}"
      on:click="_openMenu(_isOpen)"
    >
      <svg class="navbar-icon">
        <use xlink:href="{_isOpen ? '#x' : '#menu'}"/>
      </svg>
    </button>

    <a href="/" class="navbar-logo-link" title="Home">
      <slot>
        <svg class="navbar-logo">
          <use xlink:href="#logo"/>
        </svg>
      </slot>
    </a>

    <!-- htmlmin:ignore -->
    <div class="navbar-links {_isOpen ? 'df' : 'dn'}"><!-- htmlmin:ignore -->
      <a
        href="/"
        class="navbar-link l-dn{current === undefined ? ' navbar-link-active' : ''}"
      >
        Home
      </a>

      {#each menuItems as _item}
        <NavbarItem _item="{_item}" current="{current}"/>
      {/each}
    </div>
  </nav>
</header>

<script>
  import NavbarItem from './NavbarItem.html';

  export default {
    components: {
      NavbarItem,
    },
    data: () => ({
      _isOpen: false,
      _hasScrolled: false,

      /* required props */
      // menuItems: [],
      // current: '', // current URL path; in sapper, pass in "props.path"
    }),
    oncreate() {
      let wait = false;
      let lastHasScrolled = false;

      const update = () => {
        const _hasScrolled = window.pageYOffset !== 0;

        // don't set component state if nothing has changed
        /* istanbul ignore else */
        if (_hasScrolled !== lastHasScrolled) {
          lastHasScrolled = _hasScrolled;
          this.set({ _hasScrolled });
        }

        wait = false;
      };

      // debounce scroll event using rAF
      const scrollHandler = () => {
        /* istanbul ignore if */
        if (wait) return;
        wait = true;
        requestAnimationFrame(update);
      };

      document.addEventListener('scroll', scrollHandler, false);
    },
    methods: {
      _openMenu(openState) {
        if (!openState) {
          this.set({ _isOpen: true });

          const clickHandler = () => {
            this.set({ _isOpen: false });
            document.removeEventListener('click', clickHandler);
          };

          // re-queue to the end of the execution queue to avoid race condition
          setTimeout(() => {
            // close when user clicks anywhere
            document.addEventListener('click', clickHandler);
          }, 0);
        }
      },
    },
  };
</script>

<style type="text/postcss">
  @import "./_variables.css";
  @import "import.css";

  /* offset page content so it's not covered by the fixed navbar */
  :global(body) {
    margin-top: $navbar-body-offset;
  }

  .navbar {
    position: fixed;
    top: 0;
    right: 0;
    left: 0;
    z-index: $navbar-zindex;
    background-color: $navbar-background-color;

    &::after {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      pointer-events: none;
      content: "";
      box-shadow: $navbar-shadow;
      opacity: 0;
      transition: opacity $navbar-animate-speed-out ease-in;
      backface-visibility: hidden; /* promote to own layer to avoid repaints */

      @if $navbar-optimize {
        will-change: opacity;
      }
    }
  }

  .navbar-active::after {
    opacity: 1;
    transition-duration: $navbar-animate-speed-in;
  }

  .navbar-button {
    padding: 0;
    margin-right: 1.2em;
    line-height: 0;
  }

  .navbar-icon {
    width: $navbar-icon-size;
    height: $navbar-icon-size;
    padding: $navbar-icon-padding;
    color: $navbar-icon-color;
  }

  .navbar-logo-link {
    line-height: 0;
  }

  .navbar-logo {
    width: $navbar-logo-size-x;
    height: $navbar-logo-size-y;
  }

  .navbar-links {
    flex-basis: 100%;
    flex-direction: column;
    flex-grow: 1;
    padding: $navbar-links-padding;
    margin: 0 -1rem;

    &.df {
      border-top: 1px solid $navbar-border-color;
    }

    @media $(l) {
      display: flex;
      flex-basis: auto;
      flex-direction: row;
      flex-grow: 0;
      padding: 0;
      margin: 0 -1rem 0 auto;
    }
  }

  /* the following 2 selectors also appear in NavbarItem */
  :global(.navbar-link) {
    padding: $navbar-link-padding;
    margin: 0;
    color: $navbar-link-color;
  }

  :global(.navbar-link-active) {
    font-weight: $text-weight-heavy;
  }

  /**
  * This is a workaround when clicking on a link with a internal target location
  * (a link to another place within the same page). Since the target element is
  * moved to the top of the page it becomes covered by the fixed navbar. We get
  * around this using an invisible pseudo element with an offset position which
  * is attached to any heading with an id.
  *
  * Only add an id property to a heading when truly necessary to avoid the small
  * CSS rendering performance hit.
  */
  @if $navbar-hash-link-fix {
    :global(h1),
    :global(h2),
    :global(h3),
    :global(h4),
    :global(h5),
    :global(h6) {
      &[id]::before {
        display: block;
        height: $navbar-hash-link-offset;
        margin-top: calc(-1 * $navbar-hash-link-offset);
        visibility: hidden;
        content: "";
      }
    }
  }
</style>
