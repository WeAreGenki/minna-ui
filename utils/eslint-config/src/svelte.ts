/**
 * Minna UI Svelte & Sapper ESLint config add-on.
 * @fileoverview Provides extra config for Svelte & Sapper projects. This
 * config is and add-on which should extend the base config preset.
 * @see https://eslint.org/docs/user-guide/configuring
 */

/* eslint-disable sort-keys */

interface BlockAttributes {
  type: string;
}

export = {
  plugins: ['svelte3'],
  settings: {
    'svelte3/ignore-styles': (attr: BlockAttributes) =>
      attr.type === 'text/postcss',
  },
  overrides: [
    // Svelte components
    {
      files: ['*.svelte'],
      rules: {
        // import order cannot be determined correctly in .svelte components
        'import/first': 'off',
        // components are compiled at build-time so devDependencies are OK
        'import/no-extraneous-dependencies': [
          'error',
          { devDependencies: true },
        ],
        // Svelte abuses the syntax with `export let`
        'import/no-mutable-exports': 'off',
      },
    },
  ],
};
