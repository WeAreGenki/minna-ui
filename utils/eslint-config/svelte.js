/**
 * Minna UI Svelte & Sapper ESLint config add-on.
 *
 * @fileoverview Provides extra config for Svelte & Sapper projects. This
 * config is and add-on which should extend the base config preset.
 * @see https://eslint.org/docs/user-guide/configuring
 */

/* eslint-disable sort-keys */

'use strict';

/** @type {import('./types').ESLintConfig} */
module.exports = {
  plugins: ['svelte3'],
  settings: {
    /**
     * @param {import('./types').BlockAttributes} attr - Tag attributes.
     * @returns {boolean} If style block should be ignored.
     */
    'svelte3/ignore-styles': (attr) => attr.type === 'text/postcss',
    // TODO: Uncomment once false positives are handled better
    // 'svelte3/lint-template': true,
  },
  overrides: [
    // Svelte components
    {
      files: ['*.svelte'],
      rules: {
        // import order cannot be determined correctly in .svelte components
        'import/first': 'off',
        // components are compiled at build-time so devDependencies are OK
        'import/no-extraneous-dependencies': [
          'error',
          { devDependencies: true },
        ],
        // Svelte abuses the syntax with `export let`
        'import/no-mutable-exports': 'off',
      },
    },
  ],
};
