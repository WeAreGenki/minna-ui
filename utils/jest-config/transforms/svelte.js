/**
 * Jest transform for compiling Svelte components into JavaScript.
 *
 * NOTE: `svelte` is not listed as a dependency so developers can use this
 * jest preset for non-svelte projects.
 */

'use strict';

const { compile } = require('svelte/compiler'); // eslint-disable-line import/no-extraneous-dependencies

/** @type {import('../types').JestTransformer} */
exports.process = (src, filename) => {
  // Strip out <style> tags to prevent errors when unable to parse PostCSS etc.
  const re = /<style[^>]*>[\S\s]*?<\/style>/g;
  const normalised = src.replace(re, '');

  const result = compile(normalised, {
    accessors: true, // Easy access to component props for tests
    css: false,
    dev: true,
    filename,
    format: 'cjs',
    preserveWhitespace: true,
  });

  const esInterop =
    'Object.defineProperty(exports, "__esModule", { value: true });';

  return {
    code: result.js.code + esInterop,
    map: result.js.map,
  };
};
