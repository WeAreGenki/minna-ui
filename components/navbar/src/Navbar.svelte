<!--
  NAVBAR COMPONENT

  A simple fixed top navigation bar with a floating effect upon scrolling. It
  will collapse into a hamburger menu when the screen width is small.

  USAGE:
    The "items" property should be an array of objects which look like this:

      <Navbar {segment} items="{[
        { url: '/page-one', text: 'Page One' },
        { url: '/page-two', text: 'Page Two' },
        { url: '/about', text: 'About Us' },
      ]}"/>

    The menu hamburger, menu close, and logo images are referenced using the
    inline SVG sprite with <symbol> + <use> elements technique. Because these
    images are critical to page rendering, we use this technique as it's the
    faster way to load UI critical SVGs. This component doesn't come with any
    SVG images so you need to provide your own. For example, you could include
    something structured like this after the opening <body> tag in your page:

      <svg class="dn" xmlns="http://www.w3.org/2000/svg">
        <defs>
          <symbol id="logo" viewBox="0 0 67.2 25.6">
            YOUR LOGO SVG CONTENTS
          </symbol>
          <symbol id="menu" class="icon" viewBox="0 0 24 24">
            <path d="M3 12h18M3 6h18M3 18h18" />
          </symbol>
          <symbol id="x" class="icon" viewBox="0 0 24 24">
            <path d="M18 6L6 18M6 6l12 12" />
          </symbol>
        </defs>
      </svg>

    Also add this to your CSS:

      .icon {
        fill: none;
        stroke: currentColor;
        stroke-linecap: round;
        stroke-linejoin: round;
        stroke-width: 2;
      }

    You may need to adjust size and whitespace to suite your logo. See the CSS
    variables below.

    Alternatively, you may replace the SVG logo with your own markup by passing
    the markup via a slot, that is, by adding it in between Navbar tags:

      <Navbar items="{[...]}">
        <span slot="logo" class="navbar-logo">YOUR LOGO</span>
      </Navbar>

  @format
-->

<script>
  import { onMount } from 'svelte';

  // props
  export let items = [];
  export let segment; // current URL path; in sapper, pass in "segment"

  // reactive data
  let isOpen;
  let hasScrolled;

  let wait;
  let lastHasScrolled;

  function update() {
    const scrolled = window.pageYOffset !== 0;

    // don't set component state if nothing has changed
    /* istanbul ignore else */
    if (scrolled !== lastHasScrolled) {
      lastHasScrolled = scrolled;
      hasScrolled = scrolled;
    }

    wait = false;
  }

  // debounce scroll event using rAF
  function scrollHandler() {
    /* istanbul ignore if */
    if (wait) return;
    wait = true;
    requestAnimationFrame(update);
  }

  function clickHandler() {
    isOpen = false;
    document.removeEventListener('click', clickHandler);
  }

  function openMenu() {
    if (!isOpen) {
      isOpen = true;

      // re-queue at end of execution queue to avoid race condition
      setTimeout(() => {
        // close when user clicks anywhere
        document.addEventListener('click', clickHandler);
      }, 0);
    }
  }

  // FIXME: Remove if wrapper once svelte is fixed
  if (typeof document !== 'undefined') {
    onMount(() => {
      document.addEventListener('scroll', scrollHandler, false);
    });
  }
</script>

<header class="navbar {(hasScrolled || isOpen) ? 'navbar-active' : ''}">
  <nav class="dfc fww con" role="navigation">
    <button
      type="button"
      class="navbar-button l-dn button-clear"
      aria-label="menu toggle"
      aria-expanded="{!!isOpen}"
      on:click="{openMenu}"
    >
      <svg class="navbar-icon">
        <use xlink:href="{isOpen ? '#x' : '#menu'}" />
      </svg>
    </button>

    <a href="/" class="navbar-logo-link" title="Home">
      <slot name="logo">
        <svg class="navbar-logo"><use xlink:href="#logo" /></svg>
      </slot>
    </a>

    <div class="navbar-links {isOpen ? 'df' : 'dn'}">
      <a
        href="/"
        class="navbar-link l-dn {segment === undefined ? 'navbar-link-active' : ''}"
      >
        Home
      </a>

      {#each items as item}
      <a
        href="{item.url}"
        class="navbar-link {segment === item.url ? 'navbar-link-active' : ''}"
        rel="{item.rel || ''}"
      >
        {item.text}
      </a>
      {/each}
    </div>
  </nav>
</header>

<style type="text/postcss">
  @import './_navbar.css';
  @import 'import.css';

  /* offset page content so it's not covered by the fixed navbar */
  :global(body) {
    margin-top: $navbar-body-offset;
  }

  .navbar {
    position: fixed;
    top: 0;
    right: 0;
    left: 0;
    z-index: $navbar-zindex;
    background-color: $navbar-background-color;

    &::after {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      pointer-events: none;
      content: '';
      box-shadow: $navbar-shadow;
      opacity: 0;
      transition: opacity $navbar-animate-speed-out ease-in-out;
      backface-visibility: hidden; /* promote to own layer to avoid repaints */

      @if $navbar-optimize {
        will-change: opacity;
      }
    }
  }

  .navbar-active::after {
    opacity: 1;
    transition-duration: $navbar-animate-speed-in;
  }

  .navbar-button {
    padding: 0;
    margin-right: 1.2em;
    line-height: 0;
  }

  .navbar-icon {
    width: $navbar-icon-size;
    height: $navbar-icon-size;
    padding: $navbar-icon-padding;
    color: $navbar-icon-color;
  }

  .navbar-logo-link {
    line-height: 0;
  }

  .navbar-logo {
    width: $navbar-logo-size-x;
    height: $navbar-logo-size-y;
  }

  .navbar-links {
    flex-basis: 100%;
    flex-direction: column;
    flex-grow: 1;
    padding: $navbar-links-padding;
    margin: 0 -1rem;

    &.df {
      border-top: 1px solid $navbar-border-color;
    }

    @media $l {
      display: flex;
      flex-basis: auto;
      flex-direction: row;
      flex-grow: 0;
      padding: 0;
      margin: 0 -1rem 0 auto;
    }
  }

  .navbar-link {
    padding: $navbar-link-padding;
    margin: 0;
    color: $navbar-link-color;
  }

  .navbar-link-active {
    font-weight: $text-weight-heavy;
  }

  /**
  * This is a workaround when clicking on a link with a internal target location
  * (a link to another place within the same page). Since the target element is
  * moved to the top of the page it becomes covered by the fixed navbar. We get
  * around this using an invisible pseudo element with an offset position which
  * is attached to any heading with an id.
  *
  * Only add an id property to a heading when truly necessary to avoid the small
  * CSS rendering performance hit.
  */
  /* stylelint-disable-next-line order/order */
  @if $navbar-hash-link-fix {
    :global(h1),
    :global(h2),
    :global(h3),
    :global(h4),
    :global(h5),
    :global(h6) {
      &[id]::before {
        display: block;
        height: $navbar-hash-link-offset;
        margin-top: calc(-1 * $navbar-hash-link-offset);
        visibility: hidden;
        content: '';
      }
    }
  }
</style>
