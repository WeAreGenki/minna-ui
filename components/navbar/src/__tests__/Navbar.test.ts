// TODO: Integration/UI tests using puppeteer
//  â†³ Responsive CSS

/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { tick } from 'svelte';
import Navbar from '../Navbar.svelte';
import UseSlot from './__fixtures__/UseSlot.svelte';

const items = [
  { text: 'Page One', url: 'page-one' },
  { text: 'Page Two', url: 'page-two' },
  { text: 'About Us', url: 'about' },
];

describe('Navbar component', () => {
  it('renders correctly with required props set', () => {
    expect.assertions(4);
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    expect(Array.isArray(component.$$.ctx.items)).toEqual(true);
    expect(component.$$.ctx.items).not.toHaveLength(0);
    const navbar = target.querySelector('.navbar')!;
    expect(navbar.getAttribute('navbar-active')).toBeNull();
    expect(target.innerHTML).toMatchSnapshot();
  });

  it('adds class if page is scrolled', async () => {
    expect.assertions(2);
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    // @ts-ignore - We know we're writing to a read only property
    window.pageYOffset = 50;
    const event = new UIEvent('scroll');
    window.dispatchEvent(event);
    await tick();
    expect(component.$$.ctx.hasScrolled).toEqual(true);
    expect(target.querySelector('.navbar-active')).not.toBeNull();
  });

  it('opens menu on button click', () => {
    expect.assertions(2);
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    expect(component.$$.ctx.isOpen).toBeFalsy();
    const button = target.querySelector<HTMLButtonElement>('button.navbar-button')!;
    button.click();
    expect(component.$$.ctx.isOpen).toBeTruthy();
  });

  it('closes menu on document click', () => {
    expect.assertions(2);
    jest.useFakeTimers();
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    component.$$.ctx.openMenu();
    jest.runAllTimers(); // for component setTimeout
    expect(component.$$.ctx.isOpen).toBeTruthy();
    const event = new MouseEvent('click');
    document.dispatchEvent(event);
    expect(component.$$.ctx.isOpen).toBeFalsy();
  });

  it('attaches event listener on menu open but not close', () => {
    expect.assertions(3);
    jest.useFakeTimers();
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    const spy = jest.spyOn(document, 'addEventListener');
    component.$$.ctx.openMenu();
    jest.runAllTimers();
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    const event = new MouseEvent('click');
    document.dispatchEvent(event);
    expect(component.$$.ctx.isOpen).toBeFalsy();
    expect(spy).not.toHaveBeenCalled();
    spy.mockRestore();
  });

  it("doesn't attach extra event listeners on multiple button clicks", () => {
    expect.assertions(2);
    jest.useFakeTimers();
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    const spy = jest.spyOn(document, 'addEventListener');
    const button = target.querySelector<HTMLButtonElement>('button.navbar-button')!;
    button.click();
    jest.runAllTimers();
    button.click();
    jest.runAllTimers();
    expect(component.$$.ctx.isOpen).toBeTruthy();
    button.click();
    jest.runAllTimers();
    expect(spy).toHaveBeenCalledTimes(1);
    spy.mockRestore();
  });

  it('shows correct icon and class when toggled', async () => {
    expect.assertions(4);
    const target = document.createElement('div');
    new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    const icon1 = target.querySelector<SVGUseElement>('.navbar-icon > use')!;
    const navbarLinks = target.querySelector('.navbar-links')!;
    expect(icon1.getAttribute('xlink:href')).toEqual('#menu');
    expect(navbarLinks.classList.contains('df')).toBeFalsy();
    const button = target.querySelector<HTMLButtonElement>('button.navbar-button')!;
    button.click();
    await tick();
    const icon2 = target.querySelector<SVGUseElement>('.navbar-icon > use')!;
    expect(icon2.getAttribute('xlink:href')).toEqual('#x');
    expect(navbarLinks.classList.contains('df')).toBeTruthy();
  });

  it('adds class to active menu item', () => {
    expect.assertions(2);
    const target = document.createElement('div');
    new Navbar({
      props: {
        items,
        segment: 'page-two',
      },
      target,
    });
    const linkPageTwo = target.querySelector('[href="page-two"]')!;
    const linkPageOne = target.querySelector('[href="page-one"]')!;
    expect(linkPageTwo.classList.contains('navbar-link-active')).toBeTruthy();
    expect(linkPageOne.classList.contains('navbar-link-active')).not.toBeTruthy();
  });

  it('can dynamically add menu items', () => {
    expect.assertions(3);
    const target = document.createElement('div');
    const component = new Navbar({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    expect(component.$$.ctx.items).toHaveLength(3);
    component.$set({
      items: [...items, { name: 'Page New', url: 'page-new' }],
    });
    expect(component.$$.ctx.items).toHaveLength(4);
    expect(target.innerHTML).toMatchSnapshot();
  });

  it('renders custom markup when slot content is used', () => {
    expect.assertions(2);
    const target = document.createElement('div');
    new UseSlot({
      props: {
        items,
        segment: undefined,
      },
      target,
    });
    const logo = target.querySelector('.navbar-logo-link')!;
    expect(logo.textContent).toEqual('Custom Slot Content');
    expect(target.innerHTML).toMatchSnapshot();
  });
});
