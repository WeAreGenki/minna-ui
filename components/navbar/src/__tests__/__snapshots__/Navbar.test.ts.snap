// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component can dynamically add menu items 1`] = `"<header class=\\"navbar \\"><nav class=\\"dfc fww con\\" role=\\"navigation\\"><button type=\\"button\\" class=\\"navbar-button l-dn button-clear\\" aria-label=\\"menu toggle\\" aria-expanded=\\"false\\"><svg class=\\"navbar-icon\\"><use xlink:href=\\"#menu\\"></use></svg></button> <a href=\\"/\\" class=\\"navbar-logo-link\\" title=\\"Home\\"><svg class=\\"navbar-logo\\"><use xlink:href=\\"#logo\\"></use></svg></a> <div class=\\"navbar-links dn\\"><a href=\\"/\\" class=\\"navbar-link l-dn navbar-link-active\\">Home</a> <a href=\\"page-one\\" class=\\"navbar-link \\" rel=\\"\\">Page One </a><a href=\\"page-two\\" class=\\"navbar-link \\" rel=\\"\\">Page Two </a><a href=\\"about\\" class=\\"navbar-link \\" rel=\\"\\">About Us </a></div></nav></header>"`;

exports[`Navbar component renders correctly with required props set 1`] = `"<header class=\\"navbar \\"><nav class=\\"dfc fww con\\" role=\\"navigation\\"><button type=\\"button\\" class=\\"navbar-button l-dn button-clear\\" aria-label=\\"menu toggle\\" aria-expanded=\\"false\\"><svg class=\\"navbar-icon\\"><use xlink:href=\\"#menu\\"></use></svg></button> <a href=\\"/\\" class=\\"navbar-logo-link\\" title=\\"Home\\"><svg class=\\"navbar-logo\\"><use xlink:href=\\"#logo\\"></use></svg></a> <div class=\\"navbar-links dn\\"><a href=\\"/\\" class=\\"navbar-link l-dn navbar-link-active\\">Home</a> <a href=\\"page-one\\" class=\\"navbar-link \\" rel=\\"\\">Page One </a><a href=\\"page-two\\" class=\\"navbar-link \\" rel=\\"\\">Page Two </a><a href=\\"about\\" class=\\"navbar-link \\" rel=\\"\\">About Us </a></div></nav></header>"`;

exports[`Navbar component renders custom markup when slot content is used 1`] = `"<header class=\\"navbar \\"><nav class=\\"dfc fww con\\" role=\\"navigation\\"><button type=\\"button\\" class=\\"navbar-button l-dn button-clear\\" aria-label=\\"menu toggle\\" aria-expanded=\\"false\\"><svg class=\\"navbar-icon\\"><use xlink:href=\\"#menu\\"></use></svg></button> <a href=\\"/\\" class=\\"navbar-logo-link\\" title=\\"Home\\">Custom Slot Content</a> <div class=\\"navbar-links dn\\"><a href=\\"/\\" class=\\"navbar-link l-dn navbar-link-active\\">Home</a> </div></nav></header>"`;
